_posts/2017/2017-02-20-continuous-documentation-with-ansible.md

```diff
@@ -0,0 +1,207 @@
+However there is also a property `epages_unity_use_systemd` which is apparently mutually exclusive with `epages_unity_use_docker_compose`.
+The purpose of having two different ways of starting the application could (and should) be questioned, it really just documents the circumstance, that while either of the two ways should work, one or the other might be broken while development is still ongoing.  
+But as Agile thinking suggests this should not prevent us from [automating the deployment as early as possible](http://www.theserverside.com/tip/Try-an-Agile-deployment-strategy).
+
+The file goes on with some more configuration of what seem to be application properties, and finally two options to create a shop with a given name when deploying.
+By looking at the defaults, anyone could get a good idea about the scope of the role without looking at the [tasks](http://docs.ansible.com/ansible/playbooks_intro.html#tasks-list).  
+As mentioned before, roles do not necessarily need to live in the same repository as playbooks, because Ansible supports [role dependencies](http://docs.ansible.com/ansible/playbooks_roles.html#role-dependencies).
+
+## Variables
+There is also a file `vars/main.yml` inside a role that can be used to document [variables](http://docs.ansible.com/ansible/playbooks_variables.html), but generally any role should be considered "immutable", similar to a class in OO programming. Only when it is used in a playbook will it receive its arguments.  
+Assigning role variables in playbooks is a very straight forward way to document things, as it only requires to know about playbook and the scope of the involved roles.  
+Other places to put variables include `group_vars/<groupname>.yml` and `host_vars/<inventory-hostname>.yml`. The latter should be avoided, because we are trying to get rid of host-specific configuration to begin with.  
+Group vars could be considered acceptable, but the goal should always be to encapsulate as much behaviour into roles as possible.
+Talking about hosts and groups this leads over to the next section, the inventory.
+
+## Inventory
+The [inventory](http://docs.ansible.com/ansible/intro_inventory.html) contains hosts divided into groups, and can also contain variables per group or host (discouraged, see above). When a playbook is executed using the `ansible-playbook` it usually receives the particular inventory file it should be using.  
+There are several ways how different inventory files can be used, for example
+having a `localhost` inventory file, which has localhost mapped to all relevant groups, and eventually roles.
+
+The role the inventory and group variables play in terms of documenting infrastructure is to capture the differences between environments, by assigning variables that are exposed by the roles.
+
+Example inventory file:
+```
```

Comment:
For code blocks you can use this:

{% highlight %}
  some code
{% endhighlight %}

see also https://github.com/ePages-de/epages-docs/blob/develop/helpdocs/examples.md as a reference.